internal/tui/manifests.go

```diff
@@ -0,0 +1,195 @@
+
+	case substratusManifestsMsg:
+		m.reading = completed
+
+		var n int
+		var single client.Object
+		for _, k := range m.Kinds {
+			items := msg.manifests[k]
+			if single == nil && len(items) > 0 {
+				single = items[0]
+			}
+			n += len(items)
+		}
+
+		log.Printf("Found (filtered) manifests: %v", n)
+
+		if n == 0 {
+			return m, func() tea.Msg { return fmt.Errorf("No substratus Server kinds found in *.yaml") }
+		} else if n == 1 {
+			return m, func() tea.Msg { return manifestSelectedMsg{obj: single} }
+		} else {
+			// TODO: Selector
+			return m, func() tea.Msg { return manifestSelectedMsg{obj: single} }
+		}
+
+	}
+
+	return m, nil
+}
+
+// View returns a string based on data in the model. That string which will be
+// rendered to the terminal.
+func (m manifestsModel) View() (v string) {
+	defer func() {
+		if v != "" {
+			v = m.Style.Render(v)
+		}
+	}()
+
+	if m.reading == inProgress {
+		v += "Reading manifests..."
+	}
+
+	return
+}
+
+type manifestSelectedMsg struct {
+	obj client.Object
+}
+
+type substratusManifestsMsg struct {
+	manifests map[string][]client.Object
+}
+
+func findSubstratusManifests(path, filename string) tea.Cmd {
+	return func() tea.Msg {
+		msg := substratusManifestsMsg{
+			manifests: map[string][]client.Object{},
+		}
+
+		var fp string
+		if filename != "" {
+			fp = filepath.Join(path, filename)
+			manifest, err := os.ReadFile(fp)
+			if err != nil {
+				return fmt.Errorf("reading file: %w", err)
+			}
+			if err := manifestToObjects(manifest, msg.manifests); err != nil {
+				return fmt.Errorf("reading manifests in file: %v: %w", filename, err)
+			}
+		} else {
+			if path == "" {
+				var err error
+				path, err = os.Getwd()
+				if err != nil {
+					return err
+				}
+			}
+
+			fp = filepath.Join(path, "*.yaml")
+			matches, err := filepath.Glob(fp)
+			if err != nil {
+				return err
+			}
+			for _, p := range matches {
+				manifest, err := os.ReadFile(p)
+				if err != nil {
+					return fmt.Errorf("reading file: %w", err)
+				}
+
+				if err := manifestToObjects(manifest, msg.manifests); err != nil {
+					return fmt.Errorf("reading manifests in file: %v: %w", p, err)
+				}
+			}
+		}
+		if len(msg.manifests) == 0 {
+			return fmt.Errorf("No manifests found: %v", fp)
+		}
+		return msg
+	}
+}
+
+func manifestToObjects(manifest []byte, m map[string][]client.Object) error {
+	split := bytes.Split(manifest, []byte("---\n"))
+	for _, doc := range split {
+		if strings.TrimSpace(string(doc)) == "" {
+			continue
+		}
+
+		obj, err := client.Decode(doc)
+		if err != nil {
+			return fmt.Errorf("decoding: %w", err)
+		}
+		if obj == nil {
+			return nil
+		}
+
+		switch t := obj.(type) {
+		case *apiv1.Model, *apiv1.Dataset, *apiv1.Server, *apiv1.Notebook:
+			kind := t.GetObjectKind().GroupVersionKind().Kind
+			if m[kind] == nil {
+				m[kind] = make([]client.Object, 0)
+			}
+			m[kind] = append(m[kind], obj)
+			// case *apiv1.Model:
```

Comment:
I would prefer to have these comments removed unless you feel there is a need to use this in the future?